plugins/arElasticSearchPlugin/lib/arElasticSearchIndexDecorator.class.php

```diff
@@ -17,67 +17,66 @@
  * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
  */
 
-use Elastica\Query;
-use Elastica\Response;
-use Elastica\Script\AbstractScript;
-use Elasticsearch\Endpoints\UpdateByQuery;
-
-/**
- * arElasticSearchIndexDecorator can be removed when Elastica >= 6.x. It is present to
- * provide Elastica/Index::updateByQuery().
- *
- * @author      sbreker <sbreker@artefactual.com>
- */
 class arElasticSearchIndexDecorator
 {
     protected $_instance;
 
-    public function __construct(Elastica\Index $instance)
+    protected $_indexPrefix;
+
+    public function __construct($prefix)
     {
-        $this->_instance = $instance;
+        $this->_instance = [];
```

Comment:
It feels strange to use an index decorator to manage multiple indexes. I'd remove this class and move the logic altogether to `arElasticSearchPlugin.class.php`.

Reply:
I looked at removing this at first but realized that one reason we'd kept this file was that since we introduced multiple indices, we needed to use the ES index name set in the AtoM config as the prefix for all index names, and we would be repeating a lot of code for adding in the prefix if we didn't include a function for doing this.

I've renamed this class and refactored some code to clarify what this class is for.